addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -415,6 +415,16 @@ export function getAdjacents(node, predicate = n => !!n) {
     return predicate(node) ? [...previous.reverse(), node, ...next] : [];
 }
 
+/**
+ * Returns the first ancestor of the given node that satisfy the predicate
+ *
+ * @param {Node} node
+ * @param {Function} [predicate] (node: Node) => boolean
+ */
+export function getFirstAncestor(node, predicate = n => !!n) {
+    return !node || predicate(node) ? node : getFirstAncestor(node.parentElement, predicate);
```

Comment:
We usually just do something like `ancestors(node).find(predicate)` for this. I don't think there's a particular need for this util.